DOC-4243: Document the new feature Edit activities and comments (#28)
exoplatform/exo-documentation: docs/GettingStarted.rst

1. Change the activity filter to **All Activities** or **My spaces** to view all of your activities.

   |image105|

2. Click on the pulldown menu on the top right of your activity you want to delete.
   Two entries appear: Edit and Delete.
   
   |image106|

3. Click on ``Delete`` button --> A confirmation pop up appears.

   |image107|

4. Click ``Yes`` button in the confirmation message to accept your deletion.

> .. note:: As an eXo Platform user, you can only delete your own activities. If you are manager of a space, you can delete any activities related to this space.

Reviewer: It misses a note for deleting my own comment. 
Also, administrator can edit and delete comments.

5. If you click ``Cancel`` button, nothing happens.

.. _Editing-Activities-Comments:

Editing activities/comments
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Starting from eXo Platform 5.2, you can edit an activity you posted or a comment you wrote.
 
.. _Edit_Activity:
 
Editing an activity
--------------------
 
1. Click the pulldown menu on the top right of your activity : |image135|. Two entries appear: Edit and Delete.
	
	|image136|

2. Click on ``Edit`` --> Your activity’s text appears in the editor area allowing you to make changes.

    |image137|

3. Make the needed changes and then click on ``Update`` button.

.. note:: The ``Update`` button remains disabled until at least one change is done.

4. If you click the ``Cancel`` button, your changes will be ignored.

> .. note:: Edition is only possible on written text. Attached images, files or link can’t be edited.

Reviewer: We can edit some images added by the image in the CKEditor toolbar.

Author: Like for comments

          editor allowing you to type a text message.
          
              |image139|
              
After saving the change you made on your activity, the activity creation timestamp will be updated by a new 
label under your name indicating the time of the last edit:

|image140|           
          
.. _Edit_Comment:
 
Editing a comment
------------------

1. Click on the pulldown menu at the right of the comment box : |image141|. Just like for activities, two entries appear:
   
   |image142|

2. Click on ``Edit``  --> Your comment’s text appears in the editor area allowing you to edit it.

   |image143|
   
3. Edit your comment and then click on ``Update`` button.

.. note:: The ``Update`` button remains disabled until you change the comment.

4. If you click the ``Cancel`` button, your changes will be ignored.

.. note:: Unlike activities, you can change an attached link/image to your comment.

           |image144|
           
Like for activities, after saving the change you made to your comment, a text appears near your name indicating that 
an edit has been done:

|image145| 

.. warning:: Activities and comments generated from other aplications such as:
             

> 				 - Adding a topic or replying to a forum discussion

Reviewer: - closing, opening and locking a topic too will engender an automatic message. 
- Adding a poll in a topic 
- Editing an event in the agenda
- Moving a wiki page
- Editing space description
- Connexion with a new user

3. Click **OK** in the confirmation message to accept your deletion.

.. note:: As an eXo Platform user, you can only delete your own activities. If you are manager of a space, you can delete any activities related to this space.

.. _Editing-Activities-Comments:

Editing activities/comments
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Starting from eXo Platform 5.2, you are able to edit an activity you posted or a comment you wrote.

Reviewer 2: ```suggestion
Starting from eXo Platform 5.2, you can edit an activity you posted or a comment you wrote.
```

 
.. _Edit_Activity:
 
Editing an activity
--------------------
 

> 1. Click the button |image135| on the top right of your activity. Two buttons appears: Edit and Delete.

Reviewer 2: ```suggestion
1. Click the pulldown menu on the top right of your activity : |image135|. Two entries appear: Edit and Delete.
```

	
	|image136|

> 2. Click on ``Edit`` button --> Your activity text appears in the editor area allowing you to make changes.

Reviewer 2: ```suggestion
2. Click on ``Edit`` --> Your activity’s text appears in the editor area allowing you to make changes.
```

    |image137|

3. Make the needed changes and then click on ``Update`` button.

> .. note:: The ``Update`` button still in disabled status until at leat a change is done.

Reviewer 2: ```suggestion
.. note:: The ``Update`` button remains disabled until at leat one change is done.
```

> 4. If you click the ``Cancel`` button, your changes will be ignored and your activity still the same.

Reviewer 2: ```suggestion
4. If you click the ``Cancel`` button, your changes will be ignored.
```

> .. note:: Edition is only applicable on the written text. Attached images, files or link could not be edited.

Reviewer 2: ```suggestion
.. note:: Edition is only possible on written text. Attached images, files or link can’t be edited.
```

              |image138|
          
          If the activity contains only attachments (link or files and/or images) the edit button opens the
          editor allowing you to type a text message.
          
              |image139|
              
After saving the chage you made on your activity, a text appears under your name indicating that an edit has been done:

|image140|           
          
.. _Edit_Comment:
 
Editing a comment
------------------

> 1. Click on the button |image141| in the right of the comment box. Same as for the activity, two buttons appear:

Reviewer 2: ```suggestion
1. Click on the pulldown menu at the right of the comment box : |image141|. Just like for activities, two entries appear:
```

   
   |image142|

> 2. Click on ``Edit`` button --> Your comment text appears in the editor area allowing you to edit it.

Reviewer 2: ```suggestion
2. Click on ``Edit``  --> Your comment’s text appears in the editor area allowing you to edit it.
```

   |image143|
   
3. Edit your comment and then click on ``Update`` button.

> .. note:: The ``Update`` button still disabled until you change the comment.

Reviewer 2: ```suggestion
.. note:: The ``Update`` button remains disabled until you change the comment.
```

> 4. If you click the ``Cancel`` button, your changes will be ignored and your comment still the same.

Reviewer 2: ```suggestion
4. If you click the ``Cancel`` button, your changes will be ignored.
```

> .. note:: Unlike the activity, you can change an attached link/image to your comment.

Reviewer 2: ```suggestion
.. note:: Unlike activities, you can change an attached link/image to your comment.
```

           |image144|
           

> Same as for activities, after saving the change you made to your comment, a text appears near your name indicating that 

Reviewer 2: ```suggestion
Like for activities, after saving the change you made to your comment, a text appears near your name indicating that 
```

an edit has been done:

|image145| 

.. warning:: Activities and comments generated from other aplications such as:
             

> 				 - Adding topic or reply to a forum discussion

Reviewer 2: ```suggestion
				 - Adding a topic or replying to a forum discussion
```

				 - Creating or joining a space
				 - Updating your profile
				 - Adding/Editing a wiki page
             
             are not editable, you just can delete them when you click on |image146| button.
             

>              Only activities generated following a document's upload/add in Documents application are editable, 

Reviewer 2: ```suggestion
             Only activities generated following a document upload in Documents application are editable, 
```

4. Click ``Yes`` button in the confirmation message to accept your deletion.

.. note:: As an eXo Platform user, you can only delete your own activities/comments. 

>           If you are manager of a space, you can delete any activities or comments related to this space.

Reviewer 2: ```suggestion
          If you are manager of a space, you can delete any activities or comments posted in this space.
```

Reviewer 2: @mzorai @hghorbel  can an space manager edit activities/comments in his space ?

Author: No the space's manager can just delete activities or comments of other users on his space. He can not edit them.

Author: rectified in the note

After saving the change you made on your activity, the activity creation timestamp will be updated by a new 
label under your name indicating the time of the last edit:

|image140|   

> .. note:: An administrator is allowed to edit and delete any activity or comment.        

Reviewer 2: ```suggestion
.. note:: An administrator can edit and delete activities and comments posted by other users.        
```
